test/player_unit.js

```diff
@@ -341,6 +341,19 @@ describe('Player', () => {
         expect(streamingEngine.switchTextStream).not.toHaveBeenCalled();
         expect(streamingEngine.unloadTextStream).not.toHaveBeenCalled();
       });
+
+      it('unloads text stream when captions are turned off', async () => {
```

Comment:
It seems that this test only verifies that unloadTextStream() is called, but your fix was to make sure that getCurrentTextStream() returns null after unloadTextStream().

So instead of verifying the call to unloadTextStream() happened (which is an implementation detail), I think you should be verifying the value of getCurrentTextStream() at the end (which is the observable behavior in the rest of the system).